src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/PublicPrivateDataCombiner.java

```diff
@@ -151,4 +157,61 @@ public Map<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>> buildMergedM
 
     return combineMO;
   }
+
+  public String logCollatedData() {
+    var result = new StringBuilder("{");
+
+    result.append("\"Managing Officer Data\": [");
+    var first = true;
+
+    for (var officer : this.combineMO.entrySet()) {
+      if (first) {
+        first = false;
+      } else {
+        result.append(", ");
+      }
+
+      String[] linkAsArray = officer.getValue().getLeft().getLinks().getSelf().split("/");
+      String publicMoId = linkAsArray[linkAsArray.length - 1];
+
+      result.append("{");
+      result.append("\"Public Hashed ID\": \"" + publicMoId + "\", ");
+      result.append(
+          "\"Private ID\": \"" + officer.getValue().getRight().getManagingOfficerId() + "\"");
+      result.append("}");
+    }
+    result.append("],");
+
+    result.append("\"Beneficial Owner Data\": [");
+
+    first = true;
+    for (var owner : this.combineBOs.entrySet()) {
+      if (first) {
+        first = false;
+      } else {
+        result.append(", ");
+      }
+
+      String[] linkAsArray = owner.getValue().getLeft().getLinks().getSelf().split("/");
```

Comment:
Possible this will break horribly if we don't find left or right

Reply:
Yes this is what the null checks earlier were for 

Yesterday you said this scenario would be handled in another ticket wouldn't it be best to leave this till then?